src/JSBSim.cpp

```diff
@@ -454,10 +454,20 @@ int real_main(int argc, char* argv[])
   // Dump the simulation state (position, orientation, etc.)
   FDMExec->GetPropagate()->DumpState();
   
-  if (FDMExec->GetIC()->NeedTrim()) {
-    trimmer = new JSBSim::FGTrim( FDMExec );
+  // Perform trim if requested via the initialization file
+  JSBSim::TrimMode icTrimRequested = (JSBSim::TrimMode)FDMExec->GetIC()->TrimRequested();
+  if (icTrimRequested != JSBSim::TrimMode::tNone) {
+    trimmer = new JSBSim::FGTrim( FDMExec, icTrimRequested );
     try {
+      // If PullUp requested then we need to pass in target Nlf
+      if (icTrimRequested == JSBSim::TrimMode::tPullup)
+        trimmer->SetTargetNlf(FDMExec->GetIC()->GetTargetNlfIC());
```

Comment:
Why isn't that code in lines 462-464 included in `FGTrim::FGTrim` ?
I am pretty sure scripts will ignore the Nlf target otherwise.

Reply:
I've moved the code to `FGTrim::FGTrim` which means the following will now also work which never worked previously in addition to pullup trim working from a `<trim>` element in the initialization file.

```xml
<!-- Pullup trim -->
<set name="simulation/do_simple_trim" value="4" />
```